doc: add alert on REPL from TCP socket (#54594)
nodejs/node: doc/api/repl.md

For an example of running a REPL instance over [`curl(1)`][], see:
<https://gist.github.com/TooTallNate/2053342>.

> This example is intended purely for educational purposes to demonstrate how

Reviewer: ```suggestion
All examples are intended purely for educational purposes to demonstrate how
```

Author: Is there another example that poses a security risk? I don't think so

Reviewer: I'm thinking more of a just-in-case scenario, if in the future one of them presents a risk, this could act as a catch-all. It's a nitpick, so feel free to ignore.

Author: Alright, my intention in this note is to highlight this specific example. If we mention all of them, users might think we are saying, "Don't use all of this in production."
